Mathlib/NumberTheory/NumberField/Units/Regulator.lean

```diff
@@ -287,15 +295,55 @@ theorem regulator_eq_det (w' : InfinitePlace K) (e : {w // w ≠ w'} ≃ Fin (ra
 
 open scoped Classical in
 /--
-The degree of `K` times the regulator of `K` is equal to the absolute value of the determinant of
-the matrix whose columns are `(mult w * log w (fundSystem K i))_i, w` and the column `(mult w)_w`.
+The degree of `K` times the regulator of the family `u` is equal to the absolute value of the
+determinant of the matrix whose columns are
+`(mult w * log w (fundSystem K i))_i, w` and the column `(mult w)_w`.
 -/
 theorem finrank_mul_regulator_eq_det (w' : InfinitePlace K) (e : {w // w ≠ w'} ≃ Fin (rank K)) :
     finrank ℚ K * regulator K =
       |(Matrix.of (fun i w : InfinitePlace K ↦
         if h : i = w' then (w.mult : ℝ) else w.mult * (w (fundSystem K (e ⟨i, h⟩))).log)).det| := by
   rw [regulator_eq_regOfFamily_fundSystem, finrank_mul_regOfFamily_eq_det]
 
+end regulator
+
+section index
+
+open ZLattice
+
+variable {K}
+
+theorem regOfFamily_div_regOfFamily {u v : Fin (rank K) → (𝓞 K)ˣ} (hv : IsMaxRank v)
+    (h : Subgroup.closure (Set.range u) ⊔ torsion K ≤ Subgroup.closure (Set.range v) ⊔ torsion K) :
+    regOfFamily u / regOfFamily v = (Subgroup.closure (Set.range u) ⊔ (torsion K)).relindex
+      (Subgroup.closure (Set.range v) ⊔ (torsion K)) := by
+  classical
+  by_cases hu : IsMaxRank u
+  · have : span ℤ (Set.range (basisOfIsMaxRank hu)) ≤ span ℤ (Set.range (basisOfIsMaxRank hv)) := by
+      rw [← toAddSubgroup_le, span_basisOfIsMaxRank hu, span_basisOfIsMaxRank hv,
+        ← map_logEmbedding_sup_torsion (Subgroup.closure (Set.range u)).toAddSubgroup,
+        ← map_logEmbedding_sup_torsion (Subgroup.closure (Set.range v)).toAddSubgroup,
+        ← SupHomClass.map_sup, ← SupHomClass.map_sup]
+      exact AddSubgroup.map_mono <| (OrderIso.le_iff_le Subgroup.toAddSubgroup).mpr h
+    rw [regOfFamily_of_isMaxRank hu, regOfFamily_of_isMaxRank hv, covolume_div_covolume_eq_relindex
+      _ _ this, span_basisOfIsMaxRank hu, span_basisOfIsMaxRank hv, AddSubgroup.relindex_map_map,
```

Comment:
Please avoid the line break in the middle of `covolume_div_covolume_eq_relindex _ _ this`!